pySDC/tests/test_convergence_controllers/test_InterpolateBetweenRestarts.py

```diff
@@ -0,0 +1,215 @@
+import pytest
+
+from pySDC.core.Hooks import hooks
+from pySDC.core.Lagrange import LagrangeApproximation
+from pySDC.core.Collocation import CollBase
+import numpy as np
+
+
+class LogInterpolation(hooks):
+    """
+    Log the solution when a step is supposed to be restarted as well as the interpolated solution to the new nodes and
+    the solution that ends up at the nodes after the restart.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.log_u_now = False
+
+    def pre_iteration(self, step, level_number):
+        if self.log_u_now:
+            L = step.levels[level_number]
+
+            self.add_to_stats(
+                process=step.status.slot,
+                time=L.time,
+                level=L.level_index,
+                iter=step.status.iter,
+                sweep=L.status.sweep,
+                type='u_inter',
+                value=L.u.copy(),
+            )
+            self.add_to_stats(
+                process=step.status.slot,
+                time=L.time,
+                level=L.level_index,
+                iter=step.status.iter,
+                sweep=L.status.sweep,
+                type='nodes_inter',
+                value=L.sweep.coll.nodes * L.params.dt,
+            )
+            self.log_u_now = False
+
+    def post_step(self, step, level_number):
+        super().post_iteration(step, level_number)
+
+        L = step.levels[level_number]
+
+        if step.status.restart:
+            self.log_u_now = True
+            self.add_to_stats(
+                process=step.status.slot,
+                time=L.time,
+                level=L.level_index,
+                iter=step.status.iter,
+                sweep=L.status.sweep,
+                type='u_before_interpolation',
+                value=L.u.copy(),
+            )
+            self.add_to_stats(
+                process=step.status.slot,
+                time=L.time,
+                level=L.level_index,
+                iter=step.status.iter,
+                sweep=L.status.sweep,
+                type='nodes',
+                value=L.sweep.coll.nodes * L.params.dt,
+            )
+
+            # double check
+            nodes_old = L.sweep.coll.nodes.copy()
+            nodes_new = L.sweep.coll.nodes.copy() * L.status.dt_new / L.params.dt
+            interpolator = LagrangeApproximation(points=np.append(0, nodes_old))
+            self.add_to_stats(
+                process=step.status.slot,
+                time=L.time,
+                level=L.level_index,
+                iter=step.status.iter,
+                sweep=L.status.sweep,
+                type='u_inter_double_check',
+                value=(interpolator.getInterpolationMatrix(np.append(0, nodes_new)) @ L.u[:])[:],
+            )
+
+
+@pytest.mark.base
+def test_InterpolateBetweenRestarts(plotting=False):
```

Comment:
Those test only check residuum after restart and iteration, if I understand correctly ?

It would be good to also check that the order corresponds, that is :

- overwrite the M-node solution on one step with order M-1 polynomial
- interpolate on a smaller (or bigger) time step, and verify that the interpolation is exact

Reply:
I don't really understand what you want here, to be honest. The tests make sure that the residual is smaller than usual after a restart, which means the interpolated values must have been a better initial guess than the usual `spread`.

I can test this more finely with what you suggested if you can elaborate a bit more. How do I verify that the interpolation is exact? Do I need to do a Dahlquist problem for that?

Reply:
Ok. I did something weird, but I think this does what you wanted...? :D

Reply:
Seems like it, nice ! Minor detail tho : you should probably use `numpy.polynomial.Polynomial` instead of `numpy.poly1d`, see https://numpy.org/doc/stable/reference/routines.polynomials.html